### DISCUSSION TITLE
Out of memory even 20% of the memory used.

### DISCUSSION THREAD

**Discussion by jiecaix** [BLAST-WarpX/warpx#4501] [Q&A] [answered]
I'm running a 3d WarpX simulation, but the Slurm system report a error of "out of memory", like following
<img width="469" alt="image" src="https://github.com/ECP-WarpX/WarpX/assets/56915687/1f0d336d-2912-4b26-af55-e2f92c74fc08">
and I add some node but it still crashed
<img width="474" alt="image" src="https://github.com/ECP-WarpX/WarpX/assets/56915687/9e80012b-af1a-4f3e-aaec-e035dca3363f">

**Comment by ax3l** [comment]
Hi @zhazhajust , here is a script to estimate the memory use:
https://github.com/ECP-WarpX/WarpX/pull/3296

What do you estimate is the amount of memory you need for this sim?
Likely, you need to parallelize over more compute nodes.

Is this a CPU or GPU simulation?

**Reply by jiecaix** [reply, marked as answer]
Hi @ax3l , Thank you for help.

Unfortunely, i'm using refinement grid so the script in https://github.com/ECP-WarpX/WarpX/pull/3296 don't work for me.

But i findout that the question maybe was caused by my coarse grid division.

## What do you estimate is the amount of memory you need for this sim?

I didn't estimate it. But i check the simulation grid that i divided, and findout that i only set a scalar max_grid parameter. It will make the grid size too big in level 0 and too much grid number to have in level 1.

```
Grids Summary:
  Level 0   38 grids  26304512 cells  100 % of domain
            smallest grid: 16 x 208 x 208  biggest grid: 16 x 208 x 208
  Level 1   432 grids  77316096 cells  4.592611336 % of domain
            smallest grid: 32 x 32 x 32  biggest grid: 64 x 64 x 64
```

But when i set smaller max_grid_size, the guard cells number will more than valid cells number, so i don't know how to calculate the "valid cells number" and "guard cells number" to make the simulation valid and in a optimazed cells number.

## Is this a CPU or GPU simulation?

It is a CPU simulation.

**Comment by jiecaix** [comment]
By the way, is there anyway such as a test mode to check the input file could be running correctly?
